RepairShop-Android/app/src/main/java/ca/mcgill/ecse321/repairshop/BookAppointment.java

```diff
@@ -1,13 +1,261 @@
+
+                timeSlotsListView.setAdapter(timeSlotsArrayAdapter);
+                timeSlotsListView.setOnItemClickListener((adapterView, view, i, l) -> {
+                    try {
+                        timeSlot = response.getJSONObject(i);
+
+                        // Book appointment and confirm with customer
+                        toPart3();
+
+                    } catch (JSONException | UnsupportedEncodingException e) {
+                        setError(e.getMessage());
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+
+                try {
+                    setError(errorResponse.get("message").toString());
+                } catch (JSONException e) {
+                    setError(e.getMessage());
+                }
+            }
+
+        });
+
+    }
+
+    // Set up the booking confirmation - book appointment + confirm
+    private void toPart3() throws JSONException, UnsupportedEncodingException {
+
+        JSONObject body = new JSONObject();
+        body.put("startTime", timeSlot.getString("startDateTime"));
+        body.put("serviceName", service.getString("name"));
+        body.put("customerEmail", email);
+
+        // Get possible appointment times
+        HttpUtils.post(this, "api/appointment/create", token, body, new JsonHttpResponseHandler() {
+
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+
+                findViewById(R.id.book2).setVisibility(View.GONE);
+                findViewById(R.id.book3).setVisibility(View.VISIBLE);
+
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+
+                try {
+                    setError(errorResponse.get("message").toString());
+                } catch (JSONException e) {
+                    setError(e.getMessage());
+                }
+            }
+
+        });
+
+    }
+
+    // Helper to display or hide an error message
+    private void setError(String errorMessage) {
+        TextView error = findViewById(R.id.bookError);
+        error.setText(errorMessage);
+        error.setVisibility(errorMessage.equals("") ? View.GONE : View.VISIBLE);
+    }
+
+    // Helper to display service duration in hours
+    private String displayDuration(int duration) {
+        if (duration == 2) return "1 hour";
+        else return (duration / 2.0) + " hours";
+    }
+
+    // Helper to convert a timestamp format (2021-03-02T15:00:00.000+00:00) to format 2021-03-02 15:00
+    private String displayDateTime(String dateTime) {
+        return dateTime.substring(0,10) + " at " + dateTime.substring(12, 16);
```

Comment:
```suggestion
        return dateTime.substring(0,10) + " at " + dateTime.substring(11, 16);
```

We might want to start at `11` for the 2nd substring because we might have hours from `10` to `23`

Reply:
Nice catch, thanks!